services/summary.go

```diff
@@ -536,3 +537,40 @@ func (srv *SummaryService) getProjectLabelsReverseResolver(user *models.User) mo
 		return projectStrings
 	}
 }
+
+func (srv *SummaryService) GetDailyProjectStats(from, to time.Time, user *models.User) ([]models.DailyProjectStat, error) {
+	durations, err := srv.durationService.Get(from, to, user, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	dailyStats := make(map[string]map[string]time.Duration)
+	for _, d := range durations {
+		date := time.Time(d.Time).Format("2006-01-02")
+		if _, ok := dailyStats[date]; !ok {
+			dailyStats[date] = make(map[string]time.Duration)
+		}
+		dailyStats[date][d.Project] += d.Duration
+	}
+
+	result := make([]models.DailyProjectStat, 0)
+	for dateStr, projects := range dailyStats {
+		date, _ := time.Parse("2006-01-02", dateStr)
+		for project, duration := range projects {
+			result = append(result, models.DailyProjectStat{
```

Comment:
Alternatively, you could follow the "functional" approach and use `slice.Map` instead, e.g. as done [here](https://github.com/muety/wakapi/blob/1d5f51dcad86976c0f02022ac79c40d1f32aeec5/routes/compat/wakatime/v1/leaders.go#L158) and in many other places throughout the code.